main.go

```diff
@@ -61,6 +61,7 @@ type HTTPProbe struct {
 	TLSConfig              config.TLSConfig  `yaml:"tls_config"`
 	Protocol               string            `yaml:"protocol"`              // Defaults to "tcp".
 	PreferredIpProtocol    string            `yaml:"preferred_ip_protocol"` // Defaults to "ip6".
+	DisableGzipEncoding    bool              `yaml:"disable_gzip_encoding"`
```

Comment:
Very much so that I wouldn't expect it to set the header unless I asked it, that was what led me to investigate this in the first place... 
My point about Akamai was that the blackbox_exporter gets hard to use for that usecase if we force-disable gzip. And while I agree with that vendor bug (although they claim it is "working as intended"...) workarounds shouldn't be in the code, I think there is a legitimate case for having gzip supported.